lib/highscore.lua

```diff
@@ -18,15 +18,15 @@ function _hs_char_index(char)
 end
 
 function _hs_save()
-  for i=0,#_hs_records-1 do
-    if _hs_records[i+1] then
-      local addr=_hs_memory_addr+_hs_bytesize*i
-      for n=1,3 do
-        poke(addr+n-1,_hs_char_index(sub(_hs_records[i+1].name,n,n)))
-      end
-      poke2(addr+3,_hs_records[i+1].score)
-    end
-  end
+ for i,record in ipairs(_hs_records) do
```

Comment:
It never occurred to me what `ipairs` was for. I could have been using this the whole time. 🤦🏼